lib/publisher.ex

```diff
@@ -165,9 +174,28 @@ defmodule GenRMQ.Publisher do
     {:ok, conn} = connect(state)
     {:ok, channel} = Channel.open(conn)
     GenRMQ.Binding.declare_exchange(channel, config[:exchange])
+
+    with_confirmations = Keyword.get(config, :activate_confirmations, false)
+    :ok = activate_confirmations(channel, with_confirmations)
     {:ok, %{channel: channel, module: module, config: config, conn: conn}}
   end
 
+  defp activate_confirmations(channel, true), do: AMQP.Confirm.select(channel)
+  defp activate_confirmations(_, _), do: :ok
+
+  defp wait_for_confirmation(channel, config) do
+    with_confirmations = Keyword.get(config, :activate_confirmations, false)
+    max_wait_time = config |> Keyword.get(:max_confirmation_wait_time, 5_000)
+    wait_for_confirmation(channel, with_confirmations, max_wait_time)
+  end
+
+  defp wait_for_confirmation(channel, true, max_wait_time), do: AMQP.Confirm.wait_for_confirms(channel, max_wait_time)
```

Comment:
Can we re-order these two so that we pattern match on the falsy clause first? Only reason I would prefer it is because that is the default, ie no wait for confirms.

Reply:
Sure, makes sense.

Reply:
Re-ordered